app/assets/javascripts/discourse/components/pwa-install-banner.js.es6

```diff
@@ -0,0 +1,53 @@
+import {
+  default as computed,
+  on
+} from "ember-addons/ember-computed-decorators";
+
+const userDismissedPromptKey = "dismissed-pwa-install-banner";
+
+export default Ember.Component.extend({
+  deferredInstallPromptEvent: null,
+
+  @on("init")
+  _registerListener() {
+    var that = this;
+    window.addEventListener("beforeinstallprompt", e => {
```

Comment:
Is this event listener being removed somewhere ? I'm afraid it's leaking.

Also you are setting a property to an even, which probably holds a reference to a DOM object and is probably going to leak too. We should ensure this is unset on destruction of the component.